src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -239,21 +243,30 @@ class BreadcrumbsContainer extends React.Component<Props, State> {
     );
   };
 
-  handleFilterBySearchTerm = (value = '') => {
-    const {filteredByFilter} = this.state;
+  handleFilterBySearchTerm = (value: string) => {
+    const {filteredByCustomSearch} = this.state;
 
     const searchTerm = value.toLocaleLowerCase();
 
-    const filteredBreadcrumbs = filteredByFilter.filter(
-      item =>
-        !!['category', 'message', 'level', 'timestamp'].find(prop => {
-          const searchValue = item[prop];
-          if (searchValue) {
-            return searchValue.toLowerCase().indexOf(searchTerm) !== -1;
-          }
-          return false;
-        })
-    );
+    const filteredBreadcrumbs = filteredByCustomSearch.filter(obj => {
+      return Object.keys(
+        pick(obj, ['type', 'category', 'message', 'level', 'timestamp', 'data'])
+      ).some(key => {
+        if (typeof obj[key] === 'string' || typeof obj[key] === 'number') {
+          return String(obj[key])
+            .toLocaleLowerCase()
+            .includes(searchTerm);
```

Comment:
nit but you use both `includes()` and `indexOf()` below, we could be consistent there.